### PR TITLE
clean up unused code in mo_photo.F90 and refactor cloud_mod 

### PR DESCRIPTION
**Pull request by tangshuaiqi**
clean up mo_photo.F90 and remove unused code and subroutines, including those related to flags `do_jshort`,  `waccm `and `xactive_prates `(all of them are false).

refactor subroutine cloud_mod: combine some loops into one loop, add variable descriptions and units, and some other simple cleanups.

### CODE REVIEW

File: components/eam/src/chemistry/mozart/mo_photo.F90 [hunk 1]
```diff
@@ -18,11 +18,10 @@ module mo_photo
 
   private
 
-  public :: photo_inti, table_photo, xactive_photo
+  public :: photo_inti, table_photo
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_photo.F90:18]: Would you please go over the code to remove unused variables using GitHub as we discussed? Following are some of the variables that I noticed but there may be more:
pverp, pi, boltz, "use phys_control, only : waccmx_is", kg2g

File: components/eam/src/chemistry/mozart/mo_photo.F90 [hunk 2]
```diff
@@ -649,49 +566,16 @@ subroutine table_photo( photos, pmid, pdel, temper, zmid, zint, &
...
-    if ((.not.do_jshort) .or. (waccm)) then
-       n_jshrt_levs = pver
-       p1 = 1 
-       p2 = pver
-    else
-       n_jshrt_levs = pver+1
-       p1 = 2
-       p2 = pver+1
-    endif
+    n_jshrt_levs = pver
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_photo.F90:485]: This routine has some single-letter variables, I think we should clean them up.

File: components/eam/src/chemistry/mozart/mo_photo.F90 [hunk 3]
```diff
@@ -731,95 +615,6 @@ subroutine table_photo( photos, pmid, pdel, temper, zmid, zint, &
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_photo.F90:523]: Would you please check if `ion_rates_idx` is used anywhere meaningfully? If not, we can remove it and also get rid of the use of `pbuf`.

File: components/eam/src/chemistry/mozart/mo_photo.F90 [hunk 4]
```diff
@@ -1212,103 +668,92 @@ subroutine cloud_mod( zen_angle, clouds, lwc, delp, srf_alb, &
...
+    do kk = pverm,1,-1
+       if( below_tau(kk) /= 0._r8 ) then
+          below_cld(kk) = below_cld(kk) / below_tau(kk)
        else
-          below_cld(k) = below_cld(k+1)
-       end if
-    end do
+          below_cld(kk) = below_cld(kk+1)
+       endif
+    enddo
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_photo.F90:649]: These above and below tau/cld calculations are very similar. Can we factor these out in a subroutine?

File: components/eam/src/chemistry/mozart/mo_photo.F90 [hunk 5]
```diff
@@ -731,95 +615,6 @@ subroutine table_photo( photos, pmid, pdel, temper, zmid, zint, &
...
-             do m = 1,phtcnt
-                if( euv_indexer(m) > 0 ) then
-                   photos(i,:,m) = esfact * euv_prates(:,euv_indexer(m))
-                endif
-             enddo
-          endif
-
           !-----------------------------------------------------------------
           !     ... compute eff_alb and cld_mult -- needs to be before jlong
           !-----------------------------------------------------------------
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_photo.F90:528]: Please mention in/outs of `cloud_mod` and other routines.

**Review by singhbalwinder — APPROVED** [review]
Looks good! Thanks for removing a lot of unused codes! I left some comments that can be addressed in this PR or your next. Most of my comments are in the code that is not touched yet, so probably you can address those when you clean/refactor those codes.

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 2 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/mozart/mo_photo.F90:485]: will do when refactoring this subroutine

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 3 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/mozart/mo_photo.F90:523]: will do when refactoring this subroutine

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 5 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/mozart/mo_photo.F90:528]: will do when refactoring this subroutine

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 4 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/mozart/mo_photo.F90:649]: I can separated them out in a subroutine, but can't think out a way to simplify the code.

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/mozart/mo_photo.F90:18]: I will do it in this PR.

File: components/eam/src/chemistry/mozart/mo_photo.F90 (hunk 4 above)
Comment (singhbalwinder) [reply components/eam/src/chemistry/mozart/mo_photo.F90:649]: You are right. The indices are inconsistent which makes it harder to simply. Ignore my comment. It is okay as is.

**Comment by tangshuaiqi** [comment]
All tests passed. merge this PR

comp_testall_cloud_mod_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**